src/osp/scientific/kepler.h

```diff
@@ -0,0 +1,158 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2024 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#pragma once
+
+#include "osp/core/math_types.h"
+
+#include <optional>
+
+namespace osp
+{
+
+/**
+ * @brief Implementation of the analytical solution to Kepler's equation using the Classical Orbital Elements
+ */
+struct KeplerOrbit
+{
+private:
+static constexpr double KINDA_SMALL_NUMBER = 1.0E-8;
+
+public:
+
+    /**
+     * Set of elements to fully define an orbit
+     */
+    struct KeplerOrbitParams {
+        double semiMajorAxis;
+        double eccentricity;
+        double inclination;
+        double argumentOfPeriapsis;
+        double longitudeOfAscendingNode;
+        double meanAnomalyAtEpoch;
+        double gravitationalParameter;
+        double epoch;
+    };
+
+    KeplerOrbitParams m_params;
+
+    KeplerOrbit(KeplerOrbitParams const params) : m_params(params) {}
```

Comment:
`KeplerOrbitParams` is large enough that it should probably always be passed by const& to the function.